src/serverrpc.cpp

```diff
@@ -106,13 +106,20 @@ CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* pare
     /// @result {string} result.city - The server city.
     /// @result {number} result.countryId - The server country ID (see QLocale::Country).
     /// @result {string} result.welcomeMessage - The server welcome message.
+    /// @result {string} result.directoryServer - The directory server to which this server requested registration, or blank if none.
     /// @result {string} result.registrationStatus - The server registration status as string (see ESvrRegStatus and SerializeRegistrationStatus).
     pRpcServer->HandleMethod ( "jamulusserver/getServerProfile", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QString dsName = "";
+
+        if ( AT_NONE != pServer->GetDirectoryType() )
+            dsName = NetworkUtil::GetDirectoryAddress ( pServer->GetDirectoryType(), pServer->GetDirectoryAddress() );
+
         QJsonObject result{
             { "name", pServer->GetServerName() },
             { "city", pServer->GetServerCity() },
             { "countryId", pServer->GetServerCountry() },
             { "welcomeMessage", pServer->GetWelcomeMessage() },
+            { "directoryServer", dsName },
             { "registrationStatus", SerializeRegistrationStatus ( pServer->GetSvrRegStatus() ) },
```

Comment:
I'd also check GetSvrRegStatus -- it may also be undefined behaviour to call it for the AT_NONE case.  I remember having to do something in the UI code recently to get the labelling correct.

Reply:
_GetSvrRegStatus_ in **server.cpp** is a proxy to the _GetSvrReqStatus_ in **serverlist.cpp**.   A quick look shows that this state is initialized to "NOT REGISTERED".  Further, switching to AT_NONE also resets this state to NOT REGISTERED as far as I can tell.  Regardless, @dtinth did account for this in the _SerializeRegistrationStatus_ method and returns an 'undefined' literal if the server reg status falls outside one of the defined constants.  My (limited) testing has shown that switching to/from standard dir servers to custom servers to invalid servers all properly report the server status.